pkgs/applications/blockchains/lightning-terminal/default.nix

```diff
@@ -0,0 +1,76 @@
+{ pkgs ? import <nixpkgs> { system = builtins.currentSystem; },
+  lib ? pkgs.lib,
+  fetchFromGitHub ? pkgs.fetchFromGitHub,
+  fetchYarnDeps ? pkgs.fetchYarnDeps,
+  nodejs ? pkgs.nodejs,
+  yarn ? pkgs.yarn,
+  gnumake ? pkgs.gnumake,
+  go ? pkgs.go,
+  git ? pkgs.git,
+  stdenv ? pkgs.stdenv,
+  mkYarnPackage ? pkgs.mkYarnPackage,
+  fixup_yarn_lock ? pkgs.fixup_yarn_lock,
+  buildGoModule ? pkgs.buildGoModule
+}:
+
+buildGoModule rec {
+  pname = "lightning-terminal";
+  version = "0.12.4-alpha";
+  src = fetchFromGitHub {
+    owner = "lightninglabs";
+    repo = "lightning-terminal";
+    rev = "v${version}";
+    hash = "sha256-gGo9abd12zZPPTBkmEfB8BNK1ZVpea2uZNpeYRb5l00=";
+  };
+
+  app = stdenv.mkDerivation {
+    name = "lightning-app";
+    src = "${src}/app";
+
+    nativeBuildInputs = [
+      fixup_yarn_lock yarn nodejs
+    ];
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = src + "/app/yarn.lock";
+      hash = "sha256-2wp+nx0qbP0NBildsFSC76f420nCzKMOyi1zHfjJrtE=";
+    };
+
+    buildPhase = ''
+      export DISABLE_ESLINT_PLUGIN=true
+      yarn --offline build;
+      mkdir -p $out/app;
+      cp -r ./build $out/app;
+    '';
+
+    distPhase = "true";
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $offlineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+    '';
+  };
+
+  vendorHash = "sha256-MwVTmwOb0wW2e9lEbr/Pm2IhHXizwEgaw5j+x8+XA+E=";
+  ldflags = [ "-s" "-w" "-X" "github.com/lightningnetwork/lnd/build.Commit=lightning-terminal-" "-X" "github.com/lightningnetwork/lnd/build.CommitHash=" "-X" "github.com/lightningnetwork/lnd/build.GoVersion=" "-X" "github.com/lightningnetwork/lnd/build.RawTags=litd,autopilotrpc,signrpc,walletrpc,chainrpc,invoicesrpc,watchtowerrpc,neutrinorpc,peersrpc" "-X" "github.com/lightninglabs/lightning-terminal.appFilesPrefix=" "-X" "github.com/lightninglabs/lightning-terminal.Commit=" "-X" "github.com/lightninglabs/loop.Commit=v0.26.6-beta" "-X" "github.com/lightninglabs/pool.Commit=v0.6.4-beta.0.20231003174306-80d8854a0c4b" "-X" "github.com/lightninglabs/taproot-assets.Commit=v0.3.2" ];
+  subPackages = [ "cmd/litcli" "cmd/litd" ];
+  tags = [ "litd" "autopilotrpc" "signrpc" "walletrpc" "chainrpc" "invoicesrpc" "watchtowerrpc" "neutrinorpc" "peersrpc" ];
+
+  postUnpack = ''
+    cp -r ${app}/app/build/* $PWD/source/app/build
+  '';
+
+  postInstall = ''
+    cp -r $out/bin/* $out/
+  '';
+
+  meta = with lib; {
```

Comment:
If possible, this should specify a mainProgram so that the package is usable by `nix run` - this should be the binary the average user would be most likely to execute when thinking of lightning terminal

Reply:
Fixed thanks.